### DISCUSSION TITLE
Configure Live Reload Server Url

### DISCUSSION THREAD

**Discussion by jwburnside** [ionic-team/capacitor#3531] [Q&A] [answered]
In the capacitor configuration file options docs, there is a configuration option for "server" and key "url" like this:

**"server": {
    // You can make the app to load an external url (i.e. to live reload)
    "url": "http://192.168.1.33:8100",
}**

This config option is automatically added to the **capacitor.config.json** file automatically after running with the live-reload option, for instance:

 **ionic capacitor run android --l --external** 

The problem is that the server url has changed on me a couple of times when I run with livereload. This causes a couple of issues with security exceptions like allowClearText. Is there a way to set the livereload url so that it runs on the same address every time? 

Here's my capacitor version info:

Capacitor:

   Capacitor CLI   : 2.4.0
   @capacitor/core : 2.4.0

**Comment by imhoffd** [comment, marked as answer]
Since live reload generally works over Wi-Fi and the address of your computer is determined by your router, not the Ionic CLI or Capacitor CLI, this isn't something we can support as a configuration option.

However, for Android you could set up port forwarding with adb (making sure your device is plugged into your computer first):
```
adb reverse tcp:8100 tcp:8100
```

It's important to change `8100` to whichever port is being used, if necessary.

Then, with the Ionic CLI, drop the `--external` flag which should serve your app on `localhost`.

**Reply by kavyantic** [reply]
Isn't there a way to manually update the changed url inside the installed android app like we do in EXPO GO or ReactNative apps via devtools, this also allows to run a livereload build in android without adb debug bridge. It would be only required when changes to native build has done.

**Comment by psyhedeliq** [comment]
Hey @imhoffd,

If I do what you suggested with a plugged-in Android Device, it works on my new Mac (M2).
My question is how do I make it work for wireless debugging with live reload?

Strange fact is that on my old Mac (Intel) fi I run the following command it works perfect:
`ionic cap run android -l --external --consolelogs`

If I do the same on my new Mac (M2), everything is fin it loads the app but then get's stuck right there when it just opened the app and my device is also no were to be found in `adb devices`.

I tried a lot of things, but nothing worked. Can't even remember how many variations I tried.

This are some logs:

`--- adb starting (pid 4401) ---
01-24 13:31:10.771  4401 22621 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:31:10.771  4401 22621 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:31:10.771  4401 22621 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:31:10.771  4401 22621 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:31:10.771  4401 22621 I adb     : main.cpp:63 
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:31:10.778  4401 22626 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:31:10.778  4401 22621 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:31:10.780  4401 22621 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:31:10.782  4401 22621 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:31:10.782  4401 22621 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
01-24 13:31:25.400  4401 22626 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:31:25.400  4401 22626 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:31:25.405  4401 22626 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:31:25.408  4401 22621 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:31:25.410  4401 22621 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:31:25.410  4401 22621 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:31:25.415  4401 22621 I adb     : auth.cpp:435 Calling send_auth_publickey
01-24 13:31:27.297  4401 22621 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:31:42.298  4401 24053 I adb     : transport.cpp:307 192.168.1.148:38633: write thread spawning
01-24 13:31:42.298  4401 24052 I adb     : transport.cpp:335 192.168.1.148:38633: read thread spawning
01-24 13:31:42.320  4401 24052 I adb     : transport.cpp:354 192.168.1.148:38633: Received STLS packet. Stopping read thread.
01-24 13:31:42.320  4401 24054 I adb     : auth.cpp:490 t->auth_key not set before handshake
01-24 13:31:42.320  4401 24054 I adb     : transport.cpp:1734 fetching keys for transport 192.168.1.148:38633
01-24 13:31:42.320  4401 24054 I adb     : auth.cpp:495 Attempting to TLS handshake
01-24 13:31:42.325  4401 24054 I adb     : tls_connection.cpp:89 [client]: Initializing adbwifi TlsConnection
01-24 13:31:42.325  4401 24054 I adb     : tls_connection.cpp:215 [client]: Starting adbwifi tls handshake
01-24 13:31:42.566  4401 24054 I adb     : auth.cpp:513 adb_tls_set_certificate
01-24 13:31:42.566  4401 24054 I adb     : auth.cpp:533 Checking for fingerprint match [2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3]
01-24 13:31:42.566  4401 24054 I adb     : auth.cpp:542 Got SHA256 match on a key
01-24 13:31:42.585  4401 24054 I adb     : tls_connection.cpp:302 [client]: Handshake succeeded.
01-24 13:31:42.585  4401 24054 I adb     : auth.cpp:498 Handshake succeeded. Waiting for CNXN packet...
01-24 13:31:42.585  4401 24056 I adb     : transport.cpp:335 192.168.1.148:38633: read thread spawning
01-24 13:31:42.587  4401 22621 I adb     : adb.cpp:176 192.168.1.148:38633: already offline
adb server killed by remote request
01-24 13:33:01.570  4401 22621 I adb     : transport.cpp:1160 kicking transport 0x125b047d0 192.168.1.148:38633
01-24 13:33:01.570  4401 22621 I adb     : transport.cpp:407 BlockingConnectionAdapter(192.168.1.148:38633): stopping
01-24 13:33:01.570  4401 24056 E adb     : tls_connection.cpp:333 [client]: SSL_read failed [invalid library (0)]
01-24 13:33:01.570  4401 24056 I adb     : transport.cpp:339 192.168.1.148:38633: read failed: Bad file descriptor
01-24 13:33:01.570  4401 24056 I adb     : transport.cpp:1253 192.168.1.148:38633: connection terminated: read failed
01-24 13:33:01.570  4401 22621 I adb     : transport.cpp:425 BlockingConnectionAdapter(192.168.1.148:38633): stopped
01-24 13:33:01.570  4401 22621 I adb     : transport.cpp:1160 kicking transport 0x114e05050 R3CT3072V8L
01-24 13:33:01.571  4401 22621 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: requested stop
--- adb starting (pid 4850) ---
01-24 13:33:01.597  4850 27035 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:33:01.597  4850 27035 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:33:01.597  4850 27035 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:33:01.597  4850 27035 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:33:01.597  4850 27035 I adb     : main.cpp:63 
01-24 13:33:01.602  4850 27040 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:33:01.602  4850 27040 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:33:01.602  4850 27040 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:33:01.602  4850 27035 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:33:01.602  4850 27035 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:33:01.603  4850 27035 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:33:01.603  4850 27035 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:33:01.604  4850 27040 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:33:01.606  4850 27035 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:33:01.608  4850 27035 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:33:01.608  4850 27035 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:33:01.612  4850 27035 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:33:42.021  4850 27035 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:33:42.021  4850 27035 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R58N83DK6TK-eM0Bh0 not in keystore
01-24 13:33:46.874  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 65536
01-24 13:33:46.877  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 196608
01-24 13:35:39.867  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 1048576
01-24 13:35:40.187  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 1048576
01-24 13:35:40.469  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 1048576
01-24 13:35:40.514  4850 27035 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 1048576
adb server killed by remote request
01-24 13:35:48.459  4850 27035 I adb     : transport.cpp:1160 kicking transport 0x139805970 R3CT3072V8L
01-24 13:35:48.460  4850 27035 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: requested stop
--- adb starting (pid 5996) ---
01-24 13:35:48.489  5996 36504 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:35:48.489  5996 36504 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:35:48.489  5996 36504 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:35:48.489  5996 36504 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:35:48.489  5996 36504 I adb     : main.cpp:63 
01-24 13:35:48.494  5996 36508 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:35:48.494  5996 36508 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:48.494  5996 36508 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:35:48.494  5996 36504 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:35:48.495  5996 36504 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:35:48.498  5996 36508 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:35:48.502  5996 36504 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:35:48.514  5996 36504 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:35:48.514  5996 36504 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:35:48.520  5996 36504 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:35:48.723  5996 36504 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:35:48.723  5996 36504 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
adb server killed by remote request
01-24 13:35:51.501  5996 36504 I adb     : transport.cpp:1160 kicking transport 0x14261fe20 R3CT3072V8L
01-24 13:35:51.503  5996 36504 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: requested stop
--- adb starting (pid 6014) ---
01-24 13:35:51.528  6014 36844 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:35:51.528  6014 36844 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:35:51.528  6014 36844 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:35:51.528  6014 36844 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:35:51.528  6014 36844 I adb     : main.cpp:63 
01-24 13:35:51.532  6014 36850 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:35:51.532  6014 36850 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:51.532  6014 36850 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:35:51.532  6014 36844 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:35:51.532  6014 36844 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:35:51.534  6014 36844 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:35:51.534  6014 36844 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:35:51.536  6014 36850 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:35:51.538  6014 36844 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:35:51.542  6014 36844 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:35:51.542  6014 36844 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:35:51.550  6014 36844 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:35:54.951  6014 36844 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 65536
01-24 13:36:03.838  6014 36844 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:36:03.838  6014 36844 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R58N83DK6TK-eM0Bh0 not in keystore
01-24 13:36:03.838  6014 36844 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:36:03.838  6014 36844 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R58N83DK6TK-eM0Bh0 not in keystore
01-24 13:36:17.824  6014 36844 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 1048576
01-24 13:36:17.829  6014 36844 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 262144
adb server killed by remote request
01-24 13:37:02.201  6014 36844 I adb     : transport.cpp:1160 kicking transport 0x13ee1e480 R3CT3072V8L
01-24 13:37:02.202  6014 36844 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: requested stop
--- adb starting (pid 6338) ---
01-24 13:37:02.258  6338 40708 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:37:02.258  6338 40708 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:37:02.258  6338 40708 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:37:02.258  6338 40708 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:37:02.258  6338 40708 I adb     : main.cpp:63 
01-24 13:37:02.258  6338 40708 I adb     : transport_mdns.cpp:237 Openscreen mdns discovery enabled
01-24 13:37:02.267  6338 40711 I adb     : usb_osx.cpp:324 reported max packet size for R3CT3072V8L is 512
01-24 13:37:02.364  6338 40708 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:37:02.366  6338 40708 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:37:02.366  6338 40716 I adb     : transport.cpp:335 R3CT3072V8L: read thread spawning
01-24 13:37:02.366  6338 40717 I adb     : transport.cpp:307 R3CT3072V8L: write thread spawning
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:659 UDP socket created for 0.0.0.0:5353
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:170 AdbUdpSocket fd=8
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:659 UDP socket created for [0000:0000:0000:0000:0000:0000:0000:0000]:5353
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:170 AdbUdpSocket fd=10
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:274 SetMulticastOutboundInterface for index=15
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketBindFailure = "endpoint: 0.0.0.0:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:513 adb_bind failed: Address already in use
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: 0.0.0.0:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:343 adb_setsockopt failed
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: 0.0.0.0:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:343 adb_setsockopt failed
01-24 13:37:02.366  6338 40708 I adb     : udp_socket.cpp:274 SetMulticastOutboundInterface for index=15
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketBindFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:538 adb_bind failed: Address already in use
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:369 adb_setsockopt failed
01-24 13:37:02.366  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.366  6338 40708 E adb     : udp_socket.cpp:369 adb_setsockopt failed
01-24 13:37:02.367  6338 40708 I adb     : udp_socket.cpp:659 UDP socket created for [0000:0000:0000:0000:0000:0000:0000:0000]:5353
01-24 13:37:02.367  6338 40708 I adb     : udp_socket.cpp:170 AdbUdpSocket fd=8
01-24 13:37:02.367  6338 40708 I adb     : udp_socket.cpp:274 SetMulticastOutboundInterface for index=17
01-24 13:37:02.367  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketBindFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.367  6338 40708 E adb     : udp_socket.cpp:538 adb_bind failed: Address already in use
01-24 13:37:02.367  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.367  6338 40708 E adb     : udp_socket.cpp:369 adb_setsockopt failed
01-24 13:37:02.367  6338 40708 E adb     : transport_mdns.cpp:72 Encountered fatal discovery error: Failure: SocketOptionSettingFailure = "endpoint: [0000:0000:0000:0000:0000:0000:0000:0000]:5353"
01-24 13:37:02.367  6338 40708 E adb     : udp_socket.cpp:369 adb_setsockopt failed
01-24 13:37:02.367  6338 40708 I adb     : transport_mdns.cpp:187 Fallback to MdnsResponder client for discovery
01-24 13:37:02.368  6338 40708 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:37:02.368  6338 40708 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R58N83DK6TK-eM0Bh0 not in keystore
01-24 13:37:02.375  6338 40708 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:37:02.376  6338 40708 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:37:02.383  6338 40708 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:37:02.770  6338 40708 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:37:02.770  6338 40708 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
01-24 13:41:28.414  6338 40708 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:41:28.415  6338 40708 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
adb server killed by remote request
01-24 13:55:46.886  6338 40708 I adb     : transport.cpp:1160 kicking transport 0x142d04eb0 R3CT3072V8L
01-24 13:55:46.886  6338 40708 I adb     : transport.cpp:407 BlockingConnectionAdapter(R3CT3072V8L): stopping
01-24 13:55:46.886  6338 40708 I adb     : usb_osx.cpp:616 Kicking handle
01-24 13:55:46.886  6338 40716 E adb     : usb_osx.cpp:591 usb_read failed with status: e00002eb
01-24 13:55:46.886  6338 40716 I adb     : transport.cpp:339 R3CT3072V8L: read failed: Undefined error: 0
01-24 13:55:46.886  6338 40716 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: read failed
01-24 13:55:46.886  6338 40708 I adb     : transport.cpp:425 BlockingConnectionAdapter(R3CT3072V8L): stopped
--- adb starting (pid 8083) ---
01-24 13:55:46.914  8083 65932 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:55:46.914  8083 65932 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:55:46.914  8083 65932 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:55:46.914  8083 65932 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:55:46.914  8083 65932 I adb     : main.cpp:63 
01-24 13:55:46.919  8083 65936 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:55:46.919  8083 65936 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:46.919  8083 65936 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:55:46.919  8083 65932 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:55:46.919  8083 65932 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:55:46.922  8083 65936 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:55:46.924  8083 65932 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:55:46.931  8083 65932 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:55:46.931  8083 65932 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:55:46.938  8083 65932 I adb     : adb.cpp:180 R3CT3072V8L: offline
adb server killed by remote request
01-24 13:55:49.924  8083 65932 I adb     : transport.cpp:1160 kicking transport 0x15b105b60 R3CT3072V8L
01-24 13:55:49.926  8083 65932 I adb     : transport.cpp:1253 R3CT3072V8L: connection terminated: requested stop
--- adb starting (pid 8087) ---
01-24 13:55:49.948  8087 65975 I adb     : main.cpp:63 Android Debug Bridge version 1.0.41
01-24 13:55:49.948  8087 65975 I adb     : main.cpp:63 Version 34.0.5-10900879
01-24 13:55:49.948  8087 65975 I adb     : main.cpp:63 Installed as /Users/c.filip/Library/Android/sdk/platform-tools/adb
01-24 13:55:49.948  8087 65975 I adb     : main.cpp:63 Running on Darwin 23.3.0 (arm64)
01-24 13:55:49.948  8087 65975 I adb     : main.cpp:63 
01-24 13:55:49.952  8087 65980 I adb     : usb_libusb.cpp:921 libusb hotplug thread started
01-24 13:55:49.952  8087 65980 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:49.952  8087 65980 I adb     : usb_libusb.cpp:804 device connected: 1-1
01-24 13:55:49.952  8087 65975 I adb     : auth.cpp:416 adb_auth_init...
01-24 13:55:49.952  8087 65975 I adb     : auth.cpp:152 loaded new key from '/Users/c.filip/.android/adbkey' with fingerprint 2120B751E1BB8413DCF83463D21ABB201F04EA46C9DACDE14E95C41D1E7A43F3
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:819 constructed LibusbConnection for device R3CT3072V8L (1-1)
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:804 device connected: 0-1.1
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:733 ignoring device usb:0-1.1: not an adb interface. (FindInterface)
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:928 libusb hotplug: device arrived
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:804 device connected: 0-2.2
01-24 13:55:49.957  8087 65980 I adb     : usb_libusb.cpp:733 ignoring device usb:0-2.2: not an adb interface. (FindInterface)
01-24 13:55:49.959  8087 65975 I adb     : usb_libusb.cpp:651 registered new usb device 'R3CT3072V8L'
01-24 13:55:49.966  8087 65975 I adb     : transport.cpp:1734 fetching keys for transport R3CT3072V8L
01-24 13:55:49.966  8087 65975 I adb     : auth.cpp:468 Calling send_auth_response
01-24 13:55:49.975  8087 65975 I adb     : adb.cpp:180 R3CT3072V8L: offline
01-24 13:55:50.231  8087 65975 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:55:50.231  8087 65975 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore
01-24 13:55:53.367  8087 65975 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 65536
01-24 13:55:53.369  8087 65975 I adb     : usb_libusb.cpp:349 submitting zero transfer for payload length 262144
01-24 13:56:03.299  8087 65975 I adb     : adb_wifi.cpp:107 Known hosts file [/Users/c.filip/.android/adb_known_hosts.pb] does not exist...
01-24 13:56:03.299  8087 65975 I adb     : mdnsresponder_client.cpp:141 service_name=adb-R3CT3072V8L-Ilctzn not in keystore`

These are some errors I get when I have Android Studio open:
`Adb connection Error:EOF
Cannot reach ADB server, attempting to reconnect.`
`Sending Tracking request failed!`
`Error: listen EADDRNOTAVAIL: address not available fe80::c2d:353d:91f9:aa72:8100 `
`could not install *smartsocket* listener: Address already in use`

Everything works fine if I do not use live reload.

I really want to get this fixed. 🙏🏻
